modules/GKB/Release/Step.pm

```diff
@@ -263,7 +263,7 @@ sub run {
 	return unless source_code_passes_tests();
 	
 	say "Running $self->{name} pre-step tests...";
-	my @pre_step_test_errors = $self->pre_step_tests();
+	my @pre_step_test_errors = grep { defined } $self->pre_step_tests();
```

Comment:
Whats the difference in capture between the new and the original code?

Reply:
This just filters out any undefined elements of the array of pre-step test errors.  It shouldn't happen, but is a precaution to avoid any undef values sneaking in.  It also allows the array to be used in an if statement `if (@pre_step_test_errors)` without resulting in being seen as true in case one of the array elements is filled but empty.